tensorly/backend/core.py

```diff
@@ -941,45 +941,144 @@ def symeig_svd(self, matrix, n_eigenvecs=None, **kwargs):
                              % self.ndim(matrix))
 
         dim_1, dim_2 = self.shape(matrix)
-        if dim_1 <= dim_2:
-            min_dim = dim_1
-            max_dim = dim_2
-        else:
-            min_dim = dim_2
-            max_dim = dim_1
+        min_dim, max_dim = min(dim_1, dim_2), max(dim_1, dim_2)
 
         if n_eigenvecs is None:
             n_eigenvecs = max_dim
 
-        if min_dim <= n_eigenvecs:
-            if n_eigenvecs > max_dim:
-                warnings.warn('Trying to compute SVD with n_eigenvecs={0}, which '
-                              'is larger than max(matrix.shape)={1}. Setting '
-                              'n_eigenvecs to {1}'.format(n_eigenvecs, max_dim))
-                n_eigenvecs = max_dim
-            # we compute decomposition on the largest of the two to keep more eigenvecs
-            dim_1, dim_2 = dim_2, dim_1
+        if n_eigenvecs > max_dim:
+            warnings.warn('Trying to compute SVD with n_eigenvecs={0}, which '
+                          'is larger than max(matrix.shape)={1}. Setting '
+                          'n_eigenvecs to {1}'.format(n_eigenvecs, max_dim))
+            n_eigenvecs = max_dim
 
-        if dim_1 < dim_2:
+        if dim_1 > dim_2:
             S, U = self.eigh(self.dot(matrix, self.transpose(matrix)))
-            S = self.sqrt(S)
+            S = self.sqrt(self.clip(S, self.eps(S.dtype)))
```

Comment:
Is that better than the other approach we have so far?
```python
 tl.where(tl.abs(S) <= tl.finfo(S.dtype).eps, 0, 1/S)[None, :]
```
Would it create any issue by replacing something essentially 0 by something that is not? 

If there's no impact I'm happy to keep this as it results in a cleaner code :)

Reply:
There were two issues:
1. Some of the negative values were smaller than `eps`, and taking the square root would result in nans. 
2. We cannot divide U by S in the next line if S has zeros.